### PR TITLE
Add Mech-mounted multi-laser.

### PR DESCRIPTION
**Pull request by r4d6**
## About The Pull Request
Basically four Solaris Canon in one. This weapon rapidly shoot 2 laser beam for thrice the power cost.
In term of damage, an AC2 deal 26 brute damage per shot, a Solaris give 35 damage per shot for 60 power consumption, the multi-laser deal 70 burn damage for 180 power. Or a bit more than a percent of the starting power cell.

### CODE REVIEW

File: code/game/mecha/equipment/weapons/laser_weapons.dm [hunk 1]
```diff
@@ -40,6 +40,19 @@
...
+/obj/item/mecha_parts/mecha_equipment/weapon/energy/laser/heavy/auto
+	equip_cooldown = 5
+	name = "\improper Multi-laser Cannon"
+	desc = "Four Solaris Laser cannon welded together to achieve an higher firerate."
+	icon_state = "mecha_auto_laser"
+	energy_drain = 180
+	projectile = /obj/item/projectile/beam/heavylaser
+	projectiles = 8
+	projectiles_per_shot = 2
+	fire_cooldown = -5
```
Comment (Trilbyspaceclone) [inline code/game/mecha/equipment/weapons/laser_weapons.dm:52]: Do not use negitive numbers

File: code/game/mecha/equipment/weapons/laser_weapons.dm [hunk 2]
```diff
@@ -40,6 +40,19 @@
...
 	fire_sound = 'sound/weapons/lasercannonfire.ogg'
 
+/obj/item/mecha_parts/mecha_equipment/weapon/energy/laser/heavy/auto
+	equip_cooldown = 5
+	name = "\improper Multi-laser Cannon"
+	desc = "Four Solaris Laser cannon welded together to achieve an higher firerate."
+	icon_state = "mecha_auto_laser"
+	energy_drain = 180
+	projectile = /obj/item/projectile/beam/heavylaser
+	projectiles = 8
```
Comment (Trilbyspaceclone) [inline code/game/mecha/equipment/weapons/laser_weapons.dm:50]: this deals 8 x 40

File: code/game/mecha/equipment/weapons/laser_weapons.dm [hunk 3]
```diff
@@ -40,6 +40,19 @@
...
+	projectiles_per_shot = 2
+	fire_cooldown = -5
+	required_type = list(/obj/mecha)
```
Comment (Trilbyspaceclone) [inline code/game/mecha/equipment/weapons/laser_weapons.dm:53]: Do not make it fit onto all mechs

File: code/modules/research/nodes/robotics.dm [hunk 4]
```diff
@@ -272,6 +272,7 @@
 
 	unlocks_designs = list(
 							/datum/design/research/item/mecha/weapon/laser_heavy,
+							/datum/design/research/item/mecha/weapon/auto_laser,
```
Comment (Trilbyspaceclone) [inline code/modules/research/nodes/robotics.dm:275]: No

File: code/modules/research/designs/mecha_equipment.dm [hunk 5]
```diff
@@ -110,6 +110,10 @@
 	name = "CH-LC \"Solaris\" laser cannon"
 	build_path = /obj/item/mecha_parts/mecha_equipment/weapon/energy/laser/heavy
 
+/datum/design/research/item/mecha/weapon/auto_laser
```
Comment (Trilbyspaceclone) [inline code/modules/research/designs/mecha_equipment.dm:113]: No

File: code/game/mecha/equipment/weapons/laser_weapons.dm (hunk 1 above)
Comment (r4d6) [reply code/game/mecha/equipment/weapons/laser_weapons.dm:52]: Fun Fact : `fire_cooldown` is never used! At all! it is `equip_cooldown` that is used as delay between shots.

File: code/game/mecha/equipment/weapons/laser_weapons.dm (hunk 3 above)
Comment (r4d6) [reply code/game/mecha/equipment/weapons/laser_weapons.dm:53]: Fair enough.

File: code/modules/research/nodes/robotics.dm (hunk 4 above)
Comment (r4d6) [reply code/modules/research/nodes/robotics.dm:275]: Why shouldn't this weapon be made available?

File: code/modules/research/designs/mecha_equipment.dm (hunk 5 above)
Comment (r4d6) [reply code/modules/research/designs/mecha_equipment.dm:113]: I mean, what is the point of adding a weapon if it is not obtainable?

File: code/modules/research/nodes/robotics.dm (hunk 4 above)
Comment (Trilbyspaceclone) [reply code/modules/research/nodes/robotics.dm:275]: Balance and lore, this should be hand crafted out of 4 lasers not printed

File: code/game/mecha/equipment/weapons/laser_weapons.dm (hunk 1 above)
Comment (Trilbyspaceclone) [reply code/game/mecha/equipment/weapons/laser_weapons.dm:52]: Correct

File: code/modules/research/nodes/robotics.dm (hunk 4 above)
Comment (r4d6) [reply code/modules/research/nodes/robotics.dm:275]: Uh. fair enough.

File: code/game/mecha/equipment/weapons/laser_weapons.dm (hunk 1 above)
Comment (r4d6) [reply code/game/mecha/equipment/weapons/laser_weapons.dm]: Still put it to 0.

**Pull request merged**